compiler-plugin/src/test/kotlin/arrow/meta/quotes/scope/plugins/GenericPlugin.kt

```diff
@@ -0,0 +1,108 @@
+package arrow.meta.quotes.scope.plugins
+
+import arrow.meta.Meta
+import arrow.meta.Plugin
+import arrow.meta.invoke
+import arrow.meta.phases.CompilerContext
+import arrow.meta.quotes.Scope
+import arrow.meta.quotes.Transform
+import arrow.meta.quotes.classorobject.*
+import arrow.meta.quotes.declaration.DestructuringDeclaration
+import arrow.meta.quotes.element.*
+import arrow.meta.quotes.element.ValueArgument
+import arrow.meta.quotes.element.whencondition.WhenCondition
+import arrow.meta.quotes.expression.*
+import arrow.meta.quotes.expression.expressionwithlabel.*
+import arrow.meta.quotes.expression.expressionwithlabel.instanceexpressionwithlabel.ThisExpression
+import arrow.meta.quotes.expression.loopexpression.*
+import arrow.meta.quotes.modifierlist.TypeReference
+import arrow.meta.quotes.nameddeclaration.notstubbed.FunctionLiteral
+import arrow.meta.quotes.nameddeclaration.stub.typeparameterlistowner.TypeAlias
+import arrow.meta.quotes.quote
+import org.jetbrains.kotlin.psi.*
+
+open class GenericPlugin : Meta {
+
+  override fun intercept(ctx: CompilerContext): List<Plugin> = listOf(
+    "Generic Plugin" {
+      meta(
+        quote(
+          { implemented(this) },
+          { element: KtElement ->
+            Transform.replace(
+              replacing = element,
+              newDeclaration = identity() // TODO: remove default implementation
+            )
+          },
+          { transform(it) }
+        )
+      )
+    }
+  )
+
+  // TODO: remove at the end
+  private fun implemented(ktElement: KtElement): Boolean = (
+    ktElement is KtBinaryExpression
```

Comment:
Optionally replace with `when (ktElement) { is KtBlockExpression.. }` and it may read better